src/providers/Auth.js

```diff
@@ -0,0 +1,22 @@
+import React, { useState } from "react"
+
+const AuthContext = React.createContext()
+
+function AuthProvider(props) {
+  const [user, setUser] = useState({})
+
+  const login = () => {} // make a login request
+  const register = () => {} // register the user
+  const logout = () => {} // clear the token in localStorage and the user data
+
+  return (
+    <AuthContext.Provider
+      value={{ user, setUser, login, logout, register }}
```

Comment:
We will not be providing setUser function to the descendants of this Provider. setUser function will only be called in the login, logout and register functions